### DISCUSSION TITLE
Interactions between covariates

### DISCUSSION THREAD

**Discussion by Raquel-RuizDiaz** [sdmTMB/sdmTMB#195] [General]
Hello!, I have a fairly easy question:
Is it possible to include interactions between covariates in sdmTMB?

I want to create a delta_gamma model with the following configuration: 

fit_dg <- sdmTMB(
  biomass ~ 0 + s(temperature) + s(depth),
  data = data,
  mesh = mesh,
  family = delta_gamma(), #< new
  spatial = "on",
  spatiotemporal = "iid",
  time = "year",
  # silent = FALSE
)

I tried to do it like in a regular GLM changing the following part of the code:
biomass ~ 0 + s(temperature) : s(depth)
and biomass ~ 0 + s(temperature) * s(depth)

none of them worked.. 

Then, I tried removing the spline and it worked but I got errors:
Warning messages:
1: In stats::nlminb(start = tmb_obj$par, objective = tmb_obj$fn, gradient = tmb_obj$gr,  :
  NA/NaN function evaluation
2: In stats::nlminb(start = tmb_obj$par, objective = tmb_obj$fn, gradient = tmb_obj$gr,  :
  NA/NaN function evaluation

So.. I guess it is not possible?

**Comment by seananderson** [comment]
Interactions should work just fine like that with regular covariates, just not with smooths. This is the same as mgcv, which does the smoother parsing. You can let a smoother interact with another variable in several ways including:

- 2-dimensional smooths: `s(x, y)` or`t2(x, y)` (`t2` allows different wiggliness in the 2 dimensions)
- by variables: `s(x, by = y)` where y is continuous (x varies linearly with y) or y is factor (different x smoother for each y)... you likely also want + y in the formula in this case too.

Some references:
https://peerj.com/articles/6876/
https://fromthebottomoftheheap.net/2017/10/10/difference-splines-i/

**Reply by Raquel-RuizDiaz** [reply]
Thanks a lot for your quick answer!
